spec/lib/rbac/filterer_spec.rb

```diff
@@ -1286,56 +1287,143 @@ def get_rbac_results_for_and_expect_objects(klass, expected_objects)
+        context "and the current_group is not the super_administrator group" do
+          before { user.current_group = group }
+
+          it "returns users from all of the current user's groups except admins" do
+            get_rbac_results_for_and_expect_objects(User, [user, other_user])
+          end
+
+          it "returns user's groups except admins" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, [group, other_group])
+          end
         end
 
-        let!(:super_administrator_user_role) do
-          FactoryBot.create(:miq_user_role, :role => "super_administrator")
+        context "and the current_group is the super_administrator group" do
+          before { user.current_group = admin_group }
+
+          it "returns all users" do
+            get_rbac_results_for_and_expect_objects(User, User.all.to_a)
+          end
+
+          it "returns user's groups" do
+            get_rbac_results_for_and_expect_objects(MiqGroup, MiqGroup.all.to_a)
+          end
         end
+      end
+
+      context "with a self_service user" do
+        let(:user)  { FactoryBot.create(:user, :role => "user_self_service") }
+        let(:group) { user.miq_groups.detect { |g| g.description == "EvmGroup-user_self_service" } }
 
-        let(:group) do
-          FactoryBot.create(:miq_group, :tenant => default_tenant, :miq_user_role => tenant_administrator_user_role)
+        before do
+          other_user # ensure we create another user in another group
         end
 
-        let!(:user_role) do
-          FactoryBot.create(:miq_user_role, :role => "user")
+        it "returns only the current user" do
+          get_rbac_results_for_and_expect_objects(User, [user])
         end
 
-        let!(:other_group) do
-          FactoryBot.create(:miq_group, :tenant => default_tenant, :miq_user_role => user_role)
+        it "returns only the current group" do
+          get_rbac_results_for_and_expect_objects(MiqGroup, [group])
         end
 
+        context "that is in multiple groups" do
+          before { user.miq_groups << other_group }
+
+          context "and the current_group is the self_service group" do
+            before { user.current_group = group }
+
+            it "returns only the current user" do
+              get_rbac_results_for_and_expect_objects(User, [user])
+            end
+
+            it "returns only the current group" do
+              get_rbac_results_for_and_expect_objects(MiqGroup, [group])
+            end
+          end
+
+          context "and the current_group is not the self_service group" do
+            before { user.current_group = other_group }
+
+            it "returns users from all of the current user's groups" do
+              get_rbac_results_for_and_expect_objects(User, [user, other_user])
+            end
+
+            it "returns user's groups" do
+              get_rbac_results_for_and_expect_objects(MiqGroup, [other_group]) # TODO: should this be [group, other_group] ???
```

Comment:
no. I think this is right.
Per the other examples, `other_group` is the only active/current group

(yes, I wish current_group were not a concept and both were returned, but that is not the current concept)